File: src/test/scala/simulations/InternalLicenceSearchSimulation.scala
package simulations

import io.gatling.core.Predef._
import io.gatling.http.Predef._
import io.gatling.http.protocol.HttpProtocolBuilder
import scenarios.InternalSearch
import utils.{Headers, SetUp}

import scala.concurrent.duration._


class InternalLicenceSearchSimulation extends Simulation {

  val httpConfiguration: HttpProtocolBuilder = http.baseUrl(SetUp.baseURL).headers(Headers.requestHeaders)
    .disableCaching
    .disableWarmUp
    .silentResources
    .perUserNameResolution
    .maxConnectionsPerHostLikeChrome

  val search =
    InternalSearch.internalWorkerLogin.inject(atOnceUsers(SetUp.users),
          constantUsersPerSec(SetUp.rampUp) during (SetUp.rampDurationInMin minutes))
  setUp(search)
    .protocols(httpConfiguration)
    .assertions(global.failedRequests.count.is(0))
}

File: src/test/java/simulations/InternalLicenceSearchSimulation.java
package simulations;

import static io.gatling.javaapi.core.CoreDsl.*;

import io.gatling.javaapi.core.*;
import io.gatling.javaapi.http.*;


import scenarios.InternalSearch;

import java.time.Duration;

import static io.gatling.javaapi.core.CoreDsl.global;
import static io.gatling.javaapi.core.OpenInjectionStep.atOnceUsers;
import static io.gatling.javaapi.http.HttpDsl.http;
import static utils.SetUp.*;

public class InternalLicenceSearchSimulation extends Simulation {

    private static final HttpProtocolBuilder httpConfiguration = http
            .baseUrl(baseURL)
            .disableAutoReferer()
            .disableCaching()
            .disableWarmUp()
            .silentResources()
            .perUserNameResolution();

    private final PopulationBuilder search = InternalSearch.internalWorkerLogin()
            .injectOpen(
                    atOnceUsers(users),
                    constantUsersPerSec(rampUp).during(Duration.ofMinutes(rampDurationInMin))
            );

    {
        setUp(search)
                .protocols(httpConfiguration)
                .assertions(global().failedRequests().count().is(0L));
    }
}